code/modules/mob/living/carbon/xenomorph/castes/king/king.dm

```diff
@@ -44,12 +44,58 @@
 	icon_state = "king_pod"
 	density = FALSE
 	anchored = TRUE
-	var/ownerhive
+	///Which hive own this pod
+	var/ownerhive = ""
+	///What xeno was designed to be the new king
+	var/mob/living/carbon/xenomorph/future_king
 
 /obj/structure/resin/king_pod/Initialize(mapload, hivenumber)
 	. = ..()
 	ownerhive = hivenumber
-	addtimer(CALLBACK(src, .proc/try_summon_king), KING_SUMMON_TIMER_DURATION)
+	addtimer(CALLBACK(src, .proc/choose_king), KING_SUMMON_TIMER_DURATION)
+
+/obj/structure/resin/king_pod/Destroy()
+	. = ..()
+	future_king?.tracked = null
+	future_king = null
+
+/obj/structure/resin/king_pod/attack_alien(mob/living/carbon/xenomorph/X, damage_amount, damage_type, damage_flag, effects, armor_penetration, isrightclick)
+	if(X != future_king)
+		to_chat(X, "<span class='notice'>You are not the future king, you cannot use the pod!</span>")
+		return
+	if(!do_after(X, 5 SECONDS, TRUE, src))
+		return
+	try_summon_king()
+
+/**
+ * Choose at random a xeno beyond live xeno of the ownerhive, and ask them if they want to become the king
+ * If no xeno accepted to become king, give it to ghosts
+ */
+/obj/structure/resin/king_pod/proc/choose_king()
+	for(var/mob/living/carbon/xenomorph/xenomorph_alive AS in shuffle(GLOB.alive_xeno_list))
```

Comment:
Think it should prioritize older xenos, so some young runner (123) can't become a king after spawning 2 minutes ago.